crates/sncast/tests/helpers/fixtures.rs

```diff
@@ -266,35 +270,116 @@ pub fn create_test_provider() -> JsonRpcClient<HttpTransport> {
-    let src_dir = Utf8PathBuf::from(src_path);
+fn copy_dir_to_temp_dir(src_dir: impl AsRef<Utf8Path>) -> TempDir {
     let temp_dir = TempDir::new().expect("Unable to create a temporary directory");
-    let dest_dir = Utf8PathBuf::from_path_buf(temp_dir.path().to_path_buf())
-        .expect("Failed to create Utf8PathBuf from PathBuf");
 
     fs_extra::dir::copy(
-        src_dir.join("src"),
-        &dest_dir,
-        &fs_extra::dir::CopyOptions::new().overwrite(true),
+        src_dir.as_ref(),
+        temp_dir.as_ref(),
+        &fs_extra::dir::CopyOptions::new().content_only(true),
     )
-    .expect("Unable to copy the src directory");
-    fs_extra::file::copy(
-        src_dir.join("Scarb.toml"),
-        dest_dir.join("Scarb.toml"),
-        &fs_extra::file::CopyOptions::new().overwrite(true),
-    )
-    .expect("Unable to copy Scarb.toml");
+    .expect("Failed to copy directory content");
+
+    temp_dir
+}
+
+#[must_use]
+pub fn duplicate_contract_directory_with_salt(
+    src_dir: impl AsRef<Utf8Path>,
+    code_to_be_salted: &str,
+    salt: &str,
+) -> TempDir {
+    let src_dir = Utf8PathBuf::from(src_dir.as_ref());
+
+    let temp_dir = copy_dir_to_temp_dir(&src_dir);
 
+    let dest_dir = Utf8PathBuf::from_path_buf(temp_dir.path().to_path_buf())
+        .expect("Failed to create Utf8PathBuf from PathBuf");
+
+    let file_to_be_salted = "src/lib.cairo";
     let contract_code =
-        fs::read_to_string(src_dir.join("src/lib.cairo")).expect("Unable to get contract code");
-    let updated_code = contract_code.replace(to_be_salted, &(to_be_salted.to_string() + salt));
-    fs::write(dest_dir.join("src/lib.cairo"), updated_code)
+        fs::read_to_string(src_dir.join(file_to_be_salted)).expect("Unable to get contract code");
+    let updated_code =
+        contract_code.replace(code_to_be_salted, &(code_to_be_salted.to_string() + salt));
+    fs::write(dest_dir.join(file_to_be_salted), updated_code)
         .expect("Unable to change contract code");
 
     temp_dir
 }
 
+#[must_use]
+pub fn duplicate_script_directory(
+    src_dir: impl AsRef<Utf8Path>,
+    deps: Vec<impl AsRef<std::path::Path>>,
+) -> TempDir {
+    let mut deps = get_deps_map_from_paths(deps);
+
+    let src_dir = Utf8PathBuf::from(src_dir.as_ref());
+
+    let temp_dir = copy_dir_to_temp_dir(&src_dir);
+
+    let dest_dir = Utf8PathBuf::from_path_buf(temp_dir.path().to_path_buf())
+        .expect("Failed to create Utf8PathBuf from PathBuf");
+
+    let manifest_path = dest_dir.join("Scarb.toml");
+    let contents = fs::read_to_string(&manifest_path).unwrap();
+    let mut parsed_toml: Table = toml::from_str(&contents)
+        .with_context(|| format!("Failed to parse {manifest_path}"))
+        .unwrap();
+
+    let deps_toml = parsed_toml
+        .get_mut("dependencies")
+        .unwrap()
+        .as_table_mut()
+        .unwrap();
+
+    if let Some(sncast_std) = deps_toml.get_mut("sncast_std") {
```

Comment:
this is required field for all scripts, right? so as this are for tests, we can assume it must exist, meaning this whole if can be refactored

Reply:
added